todolist copy 3/main.js

```diff
@@ -0,0 +1,74 @@
+const addTaskButton = document.getElementById('addTask');
+const deskTaskInput = document.getElementById('taskInput');
+const toDoList = document.getElementById('tasklist');
+const clearing = document.getElementById('clearAll');
+
+
+let tasks = [];
+!localStorage.tasks ? tasks = [] : tasks = JSON.parse(localStorage.getItem('tasks'));

```

Comment:
We prefer not to use dot notation because when the local storage property isn't defined yet, getting it with dot notation will return "undefined." And JSON.parse fails with an "undefined" argument. So the local storage method getItem() handles this issue internally, and for not defined property returns "null." So it's a common practice to work with localStorage using its built-in methods